config-helper.js

```diff
@@ -47,7 +47,26 @@ module.exports = {
 		} else if (Array.isArray(config.entry)) {
 			config.entry[config.entry.length - 1] = replacement;
 		} else if (typeof config.entry === 'object') {
-			this.replaceMain({entry: config.entry[opts.chunk || 'main']}, replacement, opts);
+			try {
+				replacement = JSON.parse(replacement);
+			} catch (e) {
+				replacement = JSON.parse('{"main": "' + replacement + '"}');
+			}
+			if (replacement?.main !== undefined) {
+				this.replaceMain({entry: config.entry[opts.chunk || 'main']}, replacement.main, opts);
+				delete replacement.main;
+			}
+			if (Object.keys(replacement).length !== 0) {
+				config.entry = {...config.entry, ...replacement};
+				if (config.optimization.splitChunks?.chunks !== undefined) {
+					config.optimization.splitChunks.chunks = 'all';
+				} else {
+					config.optimization.splitChunks = {...config.optimization.splitChunks, chunks: 'all'};
+				}
+				config.plugins.forEach(plugin => {
+					if (plugin?.options?.ignoreOrder !== undefined) plugin.options.ignoreOrder = true;
+				});
+			}
```

Comment:
How about making new function `replaceEntry`? Because this certainly not "replacing main only". I think we could use old `replaceMain` as is.

Reply:
OK